src/Assigning/Assignment.hs

```diff
@@ -381,22 +380,55 @@ instance (Enum grammar, Eq1 ast, Ix grammar, Show grammar) => MonadError (Error
     Choose choices atEnd (Just onError) -> Tracing cs (Choose (fmap (>>= continue) choices) (fmap (>>= continue) atEnd) (Just (\ err -> (onError err >>= continue) <|> handler err))) `Then` pure
     _ -> Tracing cs assignment `Then` ((`catchError` handler) . continue)) (fmap pure rule)
 
-instance Show1 f => Show1 (Tracing f) where
-  liftShowsPrec sp sl d = liftShowsPrec sp sl d . runTracing
-
-instance (Enum grammar, Ix grammar, Show grammar, Show1 ast) => Show1 (AssignmentF ast grammar) where
```

Comment:
These `Show1` instances were likewise needed to justify the wrong, bad definition of `notFollowedBy`. But AFAIK we never, ever _use_ `notFollowedBy`, anyway.